internal/provider/common.go

```diff
@@ -9,30 +10,48 @@ import (
 	"github.com/containerd/containerd/platforms"
 	"github.com/crazy-max/diun/v4/internal/model"
 	"github.com/crazy-max/diun/v4/pkg/registry"
-	"github.com/pkg/errors"
 )
 
 var (
 	metadataKeyChars  = `a-zA-Z0-9_`
 	metadataKeyRegexp = regexp.MustCompile(`^[` + metadataKeyChars + `]+$`)
+	ErrInvalidLabel   = errors.New("invalid label error")
 )
 
+func UpdateMap(m1, m2 map[string]string) map[string]string {
+	if m1 == nil && m2 != nil {
+		return m2
+	}
+
+	if m1 != nil && m2 == nil {
+		return m1
+	}
+
+	if m1 == nil && m2 == nil {
+		return nil
+	}
+
+	for key, value := range m2 {
+		m1[key] = value
+	}
+
+	return m1
+}
```

Comment:
Cool. Didn't know about that! I had already written this for the Nomad provider and merely refactored it in this PR so I could use it elsewhere. Rather than making that change now, could refactoring both cases to use `mergo` in a subsequent PR be ok?

Reply:
Never mind. I can revert the refactor here out and use mergo for this PR leaving the migration of the nomad provider to mergo later.

Reply:
The related changes here are reverted. I cut a new PR for updating the Nomad provider (#953).

Because it's a revert here, this PR should probably be squashed before merging or merged before the other PR or else there will be a conflict.